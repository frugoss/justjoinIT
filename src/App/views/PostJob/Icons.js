import React from "react";
import styles from "../Home/home.module.scss";
import ButtonBase from "@material-ui/core/ButtonBase";


const Icons = ({name, to, img, onChange, color, background}) => {
    return (
        <div className={styles.iconContainerAdd}>
            <ButtonBase disableRipple onClick={() => {
                onChange({name: to, img, background, color})
            }}
            >
                <img className={styles.langImg} src={img} alt="Logo"/> </ButtonBase>
            <div className={styles.test}>{name}</div>
        </div>
    )
};


export default Icons;
