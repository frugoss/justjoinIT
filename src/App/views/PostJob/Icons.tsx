import React from "react";
import styles from "../Home/home.module.scss";
import ButtonBase from "@material-ui/core/ButtonBase";

type IconsProps = {
    name: string,
    to: string,
    img: string,
    onChange: (event: any) => void,
    color: string,
    background: string
}
const Icons: React.FC<IconsProps> = ({name, to, img, onChange, color, background}) => {
    return (
        <div className={styles.iconContainerAdd}>
            <ButtonBase disableRipple onClick={() =>
                onChange([{name: to, img, background, color}])
            }
            >
                <img className={styles.langImg} src={img} alt="Logo"/> </ButtonBase>
            <div className={styles.test}>{name}</div>
        </div>
    )
};


export default Icons;
