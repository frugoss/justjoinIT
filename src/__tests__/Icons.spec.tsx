import React from 'react';
import "@testing-library/jest-dom/extend-expect"
import Icons, {IconsProps} from "../App/views/PostJob/Icons";
import { render} from '@testing-library/react';

function renderIcons(props: Partial<IconsProps> = {}) {
    const defaultProps: IconsProps = {
        onChange: () => {},
        name: "test",
        img: "img",
        color: "color",
        background: "background",
        to: "test"

    };
    return render(<Icons {...defaultProps} {...props} />);
}
describe("<Icons />", () => {
    it("snapshot",  () => {
        const {container} = renderIcons();
        expect(container).toMatchSnapshot();

    })
})
