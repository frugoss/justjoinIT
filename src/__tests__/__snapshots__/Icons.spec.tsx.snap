// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icons /> snapshot 1`] = `
<div>
  <div
    class="iconContainerAdd"
  >
    <button
      class="MuiButtonBase-root"
      tabindex="0"
      type="button"
    >
      <img
        alt="Logo"
        class="langImg"
        src="img"
      />
       
    </button>
    <div
      class="test"
    >
      test
    </div>
  </div>
</div>
`;
